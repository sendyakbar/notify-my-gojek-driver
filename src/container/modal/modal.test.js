import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Modal from './modal.jsx';

describe('<Modal> component', () => {
    const wrapper = shallow(<Modal />)
    test('render correctly', () => {
        expect(wrapper).toHaveLength(1)
    })
    test('it has message state', () => {
        const messageState = wrapper.state('message')
        expect(messageState).toBeDefined()
    })
    test('it has 3 divs', () => {
        const wrapperDiv = wrapper.find('div')
        expect(wrapperDiv).toHaveLength(3)
    })
    test('it has a paragraph', () => {
        const wrapperP = wrapper.find('p')
        expect(wrapperP).toHaveLength(1)
    })
    test('renders correctly and matched snapshot', () => {
        const tree = renderer.create(<Modal />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})