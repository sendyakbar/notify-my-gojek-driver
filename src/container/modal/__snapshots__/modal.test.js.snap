// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal> component renders correctly and matched snapshot 1`] = `
<div
  className="modal"
  id="myModal"
>
  <div
    className="modal-content"
  >
    <p>
       
       
    </p>
    <div>
      <textarea
        onChange={[Function]}
        placeholder="Message"
        value=""
      />
    </div>
    <div
      className="buttonContainer"
    >
      <div>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#7BC415",
            }
          }
        >
          Send
        </button>
      </div>
      <div>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#DC6969",
            }
          }
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
